src/main/java/parser/Function.java

```diff
@@ -273,9 +273,10 @@ public static boolean assignObject(String input) {
 
                 success = true;
             } else {
-                MathExpression expr = new MathExpression(rhs);
+                MathExpression expr = mathExpClass == BigMathExpression.class ? new BigMathExpression(rhs) : new MathExpression(rhs);
                 String val = expr.solve();
                 String referenceName = expr.getReturnObjectName();
+                System.out.println("rhs: "+rhs+", mathExpClass: "+mathExpClass+", expr.class: "+expr.getClass()+", val: "+val+", type: "+expr.getReturnType());
```

Comment:
```suggestion
```

Reply:
Yah, tracing/verbose mode is a bit missing.